### PR TITLE
Add-to-cart-directive_CU-md8hxw

### PR DESCRIPTION
**Pull request by a-wilkes**

### CODE REVIEW

**Comment by a-wilkes** [comment]
Task linked: [CU-md8hxw Add to cart directive](https://app.clickup.com/t/md8hxw)

**Comment by lpboucher** [comment]
Task linked: [CU-md8hxw Add to cart directive](https://app.clickup.com/t/md8hxw)

File: src/app/shared/directives/snipcart-add.directive.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { Directive, ElementRef, Input, OnInit } from "@angular/core";
+
+import { Product } from "@shared/models/classes/product.class";
+
+@Directive({
+    selector: "[appSnipcartAdd]"
```
Comment (lpboucher) [inline src/app/shared/directives/snipcart-add.directive.ts]: i would remove the app and instead call it snipcartAdd

File: src/app/shared/directives/snipcart-add.directive.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+    constructor(private el: ElementRef) {}
+
+    ngOnInit(): void {
+        this.el.nativeElement.className += " snipcart-add-item";
+
+        this.el.nativeElement.setAttribute("data-item-id", `${ this.appSnipcartAdd.id }`);
+        this.el.nativeElement.setAttribute("data-item-url", `/products/${ this.appSnipcartAdd.productType }/${ this.appSnipcartAdd.slug }`);
+        this.el.nativeElement.setAttribute("data-item-name", `${ this.appSnipcartAdd.name }`);
+        this.el.nativeElement.setAttribute("data-item-price", `${ this.appSnipcartAdd.price }`);
```
Comment (lpboucher) [inline src/app/shared/directives/snipcart-add.directive.ts]: except for data-item-url, do you really need to a template string? all these variables are strings anyway so you can just use them directly

File: src/app/shared/directives/snipcart-add.directive.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    selector: "[appSnipcartAdd]"
+})
+export class SnipcartAddDirective implements OnInit {
+    @Input() appSnipcartAdd!: Product;
+
+    constructor(private el: ElementRef) {}
+
+    ngOnInit(): void {
+        this.el.nativeElement.className += " snipcart-add-item";
+
```
Comment (lpboucher) [inline src/app/shared/directives/snipcart-add.directive.ts:15]: for better readability, I would destructure the product here:

const { id, productType, slug, name, price } = snipcartAdd;

that way, adding the data-items will feel more natural

**Review by lpboucher — CHANGES_REQUESTED** [review]

**Pull request merged**